simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -54,16 +60,21 @@ protected ByteArrayOutputStream initialValue()
         public void handle(HttpExchange t) throws IOException {
             String query = t.getRequestURI().getRawQuery();
 
+            String contextPath = t.getHttpContext().getPath();
             ByteArrayOutputStream response = this.response.get();
             response.reset();
             OutputStreamWriter osw = new OutputStreamWriter(response);
-            TextFormat.write004(osw,
-                    registry.filteredMetricFamilySamples(parseQuery(query)));
+            if(HEALTH_PATH.equals(contextPath)) {
```

Comment:
Minor: an alternative could be to have a distinct implementation of `HttpHandler` to respond to health check. And res-use code via composition or inheritance.

Reply:
https://github.com/prometheus/client_java/issues/529
there is issue for this. This could be taken as part of this issue.